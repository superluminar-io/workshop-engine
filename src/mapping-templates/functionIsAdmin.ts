export const request = `
  #if( $ctx.identity.resolverContext.role != 'ADMIN' )
      $util.error("Unauthorized")
  #end
  {
    "version": "2018-05-29",
    "payload": {}
  }
`;

export const response = `
  $util.toJson($context.result)
`;
