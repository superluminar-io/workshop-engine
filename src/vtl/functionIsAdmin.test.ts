import Parser from 'appsync-template-tester';
import * as functionIsAdmin from './functionIsAdmin';

describe('request mapping', () => {
  describe('user has admin role', () => {
    it('should match snapshot', () => {
      const parser = new Parser(functionIsAdmin.request);

      const context = {
        identity: {
          resolverContext: {
            role: 'ADMIN',
          },
        },
      };

      expect(parser.resolve(context)).toMatchSnapshot();
    });
  });

  describe('user has not admin role', () => {
    it('should throw an error', () => {
      const parser = new Parser(functionIsAdmin.request);

      const context = {
        identity: {
          resolverContext: {
            role: 'ATTENDEE',
          },
        },
      };

      expect(() => parser.resolve(context)).toThrowError('Unauthorized');
    });
  });
});

describe('response mapping', () => {
  it('should match snapshot', () => {
    const parser = new Parser(functionIsAdmin.response);

    const context = {
      result: {
        any: 'data',
      },
    };

    expect(parser.resolve(context)).toMatchSnapshot();
  });
});
