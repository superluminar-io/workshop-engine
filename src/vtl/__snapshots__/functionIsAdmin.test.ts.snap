// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`request mapping user has admin role should match snapshot 1`] = `
Object {
  "payload": Object {},
  "version": "2018-05-29",
}
`;

exports[`response mapping should match snapshot 1`] = `
Object {
  "any": "data",
}
`;
